src/calculate/pass_4/rah.rs

```diff
@@ -0,0 +1,199 @@
+use crate::calculate::item::Object;
+use crate::calculate::pass_3::Cache;
+use crate::calculate::Ship;
+use crate::info::Info;
+
+const RAH_TYPE_ID: i32 = 4403;
+const RESISTANCE_SHIFT_AMOUNT_ID: i32 = 1849;
+const RESONANCE_IDS: [i32; 4] = [267, 268, 269, 270]; // Order: Em, Explo, Kin, Therm
```

Comment:
Please use as little static numbers as possible. These 4, and the one above, should be looked up via `info.attribute_name_to_id`.

This makes this repo more dynamic to changes in the dogma, and also makes it more verbose what those numbers refer to.

The `RAH_TYPE_ID` is more difficult; so for now that is more than fine.